### PR TITLE
fix: Fix multiple issues found by Sonatype Lift static analysis

### PR DESCRIPTION
**Pull request by odubajDT**
## This PR
- fixes some bugs found by statitc analyzer sonatype lift

### CODE REVIEW

File: distributor/pkg/lib/events/nats_connection_handler.go [hunk 1]
```diff
@@ -78,7 +78,10 @@ func (nch *NatsConnectionHandler) QueueSubscribeToTopics(topics []string, queueG
 	}
 
 	if len(topics) > 0 && !IsEqual(nch.topics, topics) {
-		nch.RemoveAllSubscriptions()
+		err := nch.RemoveAllSubscriptions()
+		if err != nil {
+			return errors.New("unable to remove all subscriptions: " + err.Error())
```
Comment (warber) [inline distributor/pkg/lib/events/nats_connection_handler.go:83]:
```suggestion
			return fmt.Errorf("unable to remove all subscriptions: %w", err)
```

File: distributor/pkg/lib/events/receiver.go [hunk 2]
```diff
@@ -62,7 +62,10 @@ func (n *NATSEventReceiver) Start(ctx *ExecutionContext) error {
 
 	defer func() {
 		ctx.Wg.Done()
-		n.natsConnectionHandler.RemoveAllSubscriptions()
+		err :=  n.natsConnectionHandler.RemoveAllSubscriptions()
+		if err != nil {
+			logger.Errorf("unable to remove all subscriptions: " + err.Error())
```
Comment (warber) [inline distributor/pkg/lib/events/receiver.go:67]:
```suggestion
			logger.Warnf("Unable to remove all subscriptions: %s", err)
```

File: jmeter-service/jmeterUtils.go [hunk 3]
```diff
@@ -134,17 +134,29 @@ func parseJMeterResult(jmeterCommandResult string, testInfo *TestInfo, workload
  * Error: error details if status was false
  */
 func executeJMeter(testInfo *TestInfo, workload *Workload, resultsDir string, url *url.URL, LTN string, funcValidation bool, logger *keptncommon.Logger) (bool, error) {
-	os.RemoveAll(resultsDir)
+	err := os.RemoveAll(resultsDir)
+	if err != nil {
+		return false, fmt.Errorf("Unable to remove path %s: %s", resultsDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:139]:
```suggestion
		return false, fmt.Errorf("unable to remove path %s: %w", err)
```

File: jmeter-service/jmeterUtils.go [hunk 4]
```diff
@@ -134,17 +134,29 @@ func parseJMeterResult(jmeterCommandResult string, testInfo *TestInfo, workload
...
+	}
 
-	os.MkdirAll(resultsDir, 0644)
+	err = os.MkdirAll(resultsDir, 0644)
+	if err != nil && resultsDir != "" {
+		return false, fmt.Errorf("Unable to create path %s: %s", resultsDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:144]:
```suggestion
		return false, fmt.Errorf("unable to create path %s: %w", resultsDir, err)
```

File: jmeter-service/jmeterUtils.go [hunk 5]
```diff
@@ -134,17 +134,29 @@ func parseJMeterResult(jmeterCommandResult string, testInfo *TestInfo, workload
...
+	}
 
 	// Step 1: Lets download all files that match /jmeter/ into a local temp directory
 	// Due to current limitations of the REST API we also fall-back and always load a specific file referenced in workload on service, stage or project level
 	// Implementing https://github.com/keptn/keptn/issues/2756
 	localTempDir := testInfo.Context
-	os.RemoveAll(localTempDir)
+	err = os.RemoveAll(localTempDir)
+	if err != nil {
+		return false, fmt.Errorf("Unable to remove path %s: %s", localTempDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:153]:
```suggestion
		return false, fmt.Errorf("unable to remove path %s: %w", localTempDir, err)
```

File: jmeter-service/jmeterUtils.go [hunk 6]
```diff
@@ -134,17 +134,29 @@ func parseJMeterResult(jmeterCommandResult string, testInfo *TestInfo, workload
...
+	}
 
-	os.MkdirAll(localTempDir, 0644)
+	err = os.MkdirAll(localTempDir, 0644)
+	if err != nil && resultsDir != "" {
+		return false, fmt.Errorf("Unable to create path %s: %s", localTempDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:158]:
```suggestion
		return false, fmt.Errorf("unable to create path %s: %w", localTempDir, err)
```

File: jmeter-service/jmeterUtils.go [hunk 7]
```diff
@@ -175,7 +187,10 @@ func executeJMeter(testInfo *TestInfo, workload *Workload, resultsDir string, ur
 	if !FileExists(mainScriptFileName) {
 		err = fmt.Errorf("JMeter script %s not found locally at %s for %s.%s.%s", workload.Script, mainScriptFileName, testInfo.Project, testInfo.Stage, testInfo.Service)
 		if removeTempFiles {
-			os.RemoveAll(localTempDir)
+			err := os.RemoveAll(localTempDir)
+			if err != nil {
+				return false, fmt.Errorf("Unable to remove path %s: %s", localTempDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:192]:
```suggestion
				return false, fmt.Errorf("unable to remove path %s: %w", localTempDir, err)
```

File: jmeter-service/jmeterUtils.go [hunk 8]
```diff
@@ -202,7 +217,10 @@ func executeJMeter(testInfo *TestInfo, workload *Workload, resultsDir string, ur
 
 	// now lets remove all downloaded files
 	if removeTempFiles {
-		os.RemoveAll(localTempDir)
+		err = os.RemoveAll(localTempDir)
+		if err != nil {
+			return false, fmt.Errorf("Unable to remove path %s: %s", localTempDir, err.Error())
```
Comment (warber) [inline jmeter-service/jmeterUtils.go:222]:
```suggestion
			return false, fmt.Errorf("unable to remove path %s: %w", localTempDir, err)
```

File: jmeter-service/main.go [hunk 9]
```diff
@@ -72,7 +72,9 @@ func gotEvent(ctx context.Context, event cloudevents.Event) error {
 // The method will always try to execute a health check workload first, then execute the workload based on the passed testStrategy
 //
 func runTests(event cloudevents.Event, shkeptncontext string, data keptnv2.TestTriggeredEventData, logger *keptncommon.Logger) {
-	sendTestsStartedEvent(shkeptncontext, event, logger)
+	if err := sendTestsStartedEvent(shkeptncontext, event, logger); err != nil {
+		logger.Error(fmt.Sprintf("Error sending test started event: %s", err.Error()))
```
Comment (warber) [inline jmeter-service/main.go:76]:
```suggestion
		logger.Warnf("unable to send test .started event: %w", err)
```

File: shipyard-controller/handler/uniformintegrationhandler.go [hunk 10]
```diff
@@ -109,7 +109,7 @@ func (rh *UniformIntegrationHandler) Register(c *gin.Context) {
 		}
 
 		raw := fmt.Sprintf("%s-%s-%s-%s-%s", integration.Name, integration.MetaData.KubernetesMetaData.Namespace, integration.Subscription.Filter.Project, integration.Subscription.Filter.Stage, integration.Subscription.Filter.Service)
-		hasher := sha1.New() //nolint:gosec
+		hasher := sha256.New() //nolint:gosec
```
Comment (warber) [inline shipyard-controller/handler/uniformintegrationhandler.go:112]: The sha1 algorithm is not used in a security sensitive context here, but just to compute a hash. Pls ignore this with `// NOSONAR`

**Review by warber — CHANGES_REQUESTED** [review]

**Comment by warber** [comment]
@odubajDT please sign your commits: see: https://github.com/keptn/keptn/pull/5934/checks?check_run_id=4140980500
Also, could you please resolve the conflicts?

**Comment by odubajDT** [comment]
> @odubajDT please sign your commits: see: https://github.com/keptn/keptn/pull/5934/checks?check_run_id=4140980500 Also, could you please resolve the conflicts?

Done, hopefully after the tests are finished, it will be ready for merging.

**Comment by odubajDT** [comment]
Codecov/patch analysis is expected to fail, as number of lines (checks for errors, small fixes,...) has increased and no new tests were added.

**Comment by odubajDT** [comment]
@bacherfl are here any suggestions? If not, can it be merged? Thank you :)

File: jmeter-service/jmeterUtils.go [hunk 11]
```diff
@@ -134,17 +156,29 @@ func parseJMeterResult(jmeterCommandResult string, testInfo *TestInfo, workload
  * Error: error details if status was false
  */
 func executeJMeter(testInfo *TestInfo, workload *Workload, resultsDir string, url *url.URL, LTN string, funcValidation bool) (bool, error) {
-	os.RemoveAll(resultsDir)
+	err := removeAllDirPath(resultsDir)
+	if err != nil {
+		return false, err
```
Comment (thisthat) [inline jmeter-service/jmeterUtils.go:161]: Would it make sense to create new test cases where all these new `return` statements of the function are checked?

Comment (odubajDT) [reply jmeter-service/jmeterUtils.go:161]: I can create unit tests for the new functions `makeAllDirPath(...)` and `removeAllDirPath(...)`, other changes are just checks of return values for errors. Is this what you are pointing at?

Comment (warber) [reply jmeter-service/jmeterUtils.go:161]: Note, that this part of jmeter service is currently experiencing some restructuring as part of #5916

Comment (odubajDT) [reply jmeter-service/jmeterUtils.go:161]: Ok, so do you prefer to wait with merging of this PR?

Comment (odubajDT) [reply jmeter-service/jmeterUtils.go:161]: Or at least we can remove the changes of jmeter service from this PR and look at it afterwards...

Comment (warber) [reply jmeter-service/jmeterUtils.go:161]: yeah sounds good

Comment (odubajDT) [reply jmeter-service/jmeterUtils.go]: @warber do you prefer removing all changes of jmeter-service or just the ones in jmeterUtils.go? Thanks

Comment (warber) [reply jmeter-service/jmeterUtils.go]: lets drop all in jmeter-service

**Review by warber — APPROVED** [review]

**Pull request merged**